### PR TITLE
ZTE-SH-CN-run-application-scale-stateful-set-2017-09-08

### PR DESCRIPTION
**Pull request by jianglingxia**
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.8 Features: set Milestone to `1.8` and Base Branch to `release-1.8`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>
> NOTE: Please check the “Allow edits from maintainers” box (see image below) to
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.

Allow edits from maintainers checkbox

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5346)

### CODE REVIEW

**Comment by jianglingxia** [comment]
cc @zhangxiaoyu-zidif

**Comment by zhangxiaoyu-zidif** [comment]
This is a translation PR from ZTE-Shanghai.

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+approvers:
+- bprashanth
+- enisoc
+- erictune
+- foxish
+- janetkuo
+- kow3ns
+- smarterclayton
+title: 伸缩StatefulSet
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: 弹缩 会不会好一些？

**Comment by chenopis** [comment]
cc @markthink

**Comment by markthink** [comment]
@jianglingxia Please close PR, update to k8smeetup library, unified submission..Do not submit a single file..

update table:
https://docs.google.com/spreadsheets/d/1FDFCv9RK5nSMgLXhPrJ5k7r5QvHnNEFnXbvoFiM8v20/edit#gid=1015862274

**Comment by markthink** [comment]
@jianglingxia Please close PR, update to k8smeetup library, unified submission..Do not submit a single file..

update table:
https://docs.google.com/spreadsheets/d/1FDFCv9RK5nSMgLXhPrJ5k7r5QvHnNEFnXbvoFiM8v20/edit#gid=1015862274

**Comment by markthink** [comment]
@chenopis  We will maintain a list of translators，As far as possible to avoid a separate submission。

https://docs.google.com/spreadsheets/d/1nHCp-wHf26lqcah9CVQpPaiSFxqVrsdFlCyRyHCjz80/edit#gid=0

**Comment by zhangxiaoyu-zidif** [comment]
@markthink 
keeping it seems to be ok.

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+```shell
+kubectl patch statefulsets <stateful-set-name> -p '{"spec":{"replicas":<new-replicas>}}'
+```
+
+## 排查故障
+
+### 缩容工作不正常
+
+当stateful管理下的任何一个Pod不健康时您不能缩容该StatefulSet. 仅当stateful下的所有Pods都处于运行和ready状态后才可缩容.
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: we can uniform caps about controller names.
How about  using Statefulset/ statefulset/ StatefulSet 
I prefer StatefulSet

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+title: 伸缩StatefulSet
+---
+
+{% capture overview %}
+本文介绍如何伸缩StatefulSet.
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* StatefulSets仅适用于Kubernetes1.5版本或高于1.5.
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: StatefulSets仅适用于Kubernetes1.5版本或高于1.5.
改为
StatefulSets仅适用于Kubernetes1.5及以上版本.

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+* **不是所有stateful applications都适合伸缩.** 在伸缩前您必须了解您的StatefulSets应用. 如果您不确定, 请记住伸缩该StatefulSets也许不安全.
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: 在伸缩前您必须了解您的StatefulSets应用. 如果您不确定, 请记住伸缩该StatefulSets也许不安全.
->
在弹缩您的应用前，您必须充分了解您的应用。不适当的弹缩StatefulSet或许会造成应用自身功能的不稳定。

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+* 仅当您确定该stateful应用的集群是完全健康才可执行伸缩操作.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## 使用 `kubectl` 伸缩StatefulSets
+
+伸缩前确保 `kubectl` 已升级至Kubernetes1.5版本或高于1.5. 如果不确定, 执行 `kubectl version` 命令并检查使用的 `Client Version`.
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: 弹缩请确认‘kubectl’已经升级到1.5及以上版本。
The version here is kubectl's, the client itself. 
Not kubernetes cluster or other components.

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+## 排查故障
+
+### 缩容工作不正常
+
+当stateful管理下的任何一个Pod不健康时您不能缩容该StatefulSet. 仅当stateful下的所有Pods都处于运行和ready状态后才可缩容.
+
+当一个StatefulSet的size > 1, 如果有一个不健康的Pod, 没有办法让Kubernetes知道是否是由于永久性故障还是瞬态(升级/维护/节点重启)导致. 如果该Pod不健康是由于永久性
+故障导致, 则在不纠正该故障的情况下进行缩容可能会导致一种状态， 即StatefulSet下的Pod数量低于应正常运行的副本数. 这也许会导致StatefulSet变成不可用.
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: 变成不可用
->
不可用

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+kubectl patch statefulsets <stateful-set-name> -p '{"spec":{"replicas":<new-replicas>}}'
+```
+
+## 排查故障
+
+### 缩容工作不正常
+
+当stateful管理下的任何一个Pod不健康时您不能缩容该StatefulSet. 仅当stateful下的所有Pods都处于运行和ready状态后才可缩容.
+
+当一个StatefulSet的size > 1, 如果有一个不健康的Pod, 没有办法让Kubernetes知道是否是由于永久性故障还是瞬态(升级/维护/节点重启)导致. 如果该Pod不健康是由于永久性
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: 如果有一个不健康的pod
->
如果有一个pod不健康

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+另外, 您可以 [in-place updates](/docs/concepts/cluster-administration/manage-deployment/#in-place-updates-of-resources) StatefulSets.
+
+如果您的StatefulSet开始由 `kubectl apply` 或 `kubectl create --save-config`创建,
+更新StatefulSet manifests中的 `.spec.replicas`, 然后执行命令 `kubectl apply`:
+
+```shell
+kubectl apply -f <stateful-set-file-updated>
+```
+
+除此之外, 可以通过命令 `kubectl edit`编辑该字段:
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: 编辑前少了一个空格

**Comment by jianglingxia** [comment]
thanks for your careful review and I have modified with your advice,If you think no problem ,I will squash into one commit,thanks! @zhangxiaoyu-zidif

File: cn/docs/tasks/run-application/scale-stateful-set.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+* StatefulSets仅适用于Kubernetes1.5及以上版本.
+* **不是所有Stateful应用都适合弹缩.** 在弹缩前您的应用前. 您必须充分了解您的应用, 不适当的弹缩StatefulSet或许会造成应用自身功能的不稳定.
+* 仅当您确定该Stateful应用的集群是完全健康才可执行弹缩操作.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## 使用 `kubectl` 伸缩StatefulSets
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/scale-stateful-set.md]: 伸缩->弹缩

Comment (jianglingxia) [reply cn/docs/tasks/run-application/scale-stateful-set.md]: thanks,I have checked all and change the '伸缩' to‘弹缩’！

**Comment by zhangxiaoyu-zidif** [comment]
LGTM

**Pull request merged**